### ISSUE TITLE
Remove or improve map interp_image

### ISSUE THREAD

**Issue by cdeil** [gammapy/gammapy#1644]
I'm currently rewriting the Fermi-LAT data with Gammapy tutorial (#1545), and was struggling a bit with the exposure cube and Galactic diffuse model cube interpolation onto my WCS geometry of choice.

I'll post an update to that notebook shortly, but I wanted to ask already whether we should remove or improve the map `interp_image` method:
https://github.com/gammapy/gammapy/blame/a61737bbc036522164d72575ea68fede7648cf64/gammapy/maps/wcsnd.py#L210

Using this example:
```
m = Map.read('$GAMMAPY_EXTRA/test_datasets/unbundled/fermi/gll_iem_v02_cutout.fits')
print(m.geom)
print(m.geom.axes[0])
```
one can indeed interpolate out an image at 1e4 MeV like this:
```
m.interp_image(coords=(1e4, ))
```
but it doesn't support coords dicts like other methods do, so e.g. this blows up with a cryptic error:
```
m.interp_image(coords={'energy': 1e4})
```
and actually what I want is to interpolate this 3D map onto another 3D map, not just interpolate an image.

For the exposure, I did use use the `interp_by_coord` instead already:
```
coord = counts.geom.get_coord()
data = exposure_hpx.interp_by_coord({
    'skycoord': coord.skycoord,
    'energy': 1000 * coord['energy'],
})
exposure = WcsNDMap(counts.geom, data, unit=exposure_hpx.unit)
```
This works fine, except that `interp_image` has built-in support to extrapolate along the energy axis, which is useful for diffuse model cubes: there you usually want to fill with 0 outside the sky image, but extrapolate on the energy axis, instead of silently filling with 0 and getting incorrect results. With `interp_by_coord` it's currently not possible to have such different extrapolation behaviour along the spatial and non-spatial dimensions, and I don't think it's easy to create it.

My suggestion would be to remove `interp_image` for now, and to try and improve `interp_by_coord` as much as possible, or add a second method later. There are currently no callers in Gammapy or Fermipy for `interp_image`. The method has no docstring and no tests, and a TODO that it should be extended to support ND, not just 3D.

@woodmd @registerrier  @adonath - Thoughts?

**Comment by adonath** [comment]
Fine by me to remove the method for now. Just a few further thoughts:

I think the current use case (extrapolating a Fermi diffuse cube) is perfectly valid and should be covered somehow in future. This involves solving the problem of defining different extra-/ interpolation behaviour for different map axes, which is currently not possible with `interp_by_coord()` and maybe can never easily be achieved, because it's not natively supported by the `RegularGridInterpolator` (which I think is why the second method exists at all...). If we want to keep the functionality or reintroduce it later the method should be renamed to `interp_image_by_coord()`, and maybe `interp_image_by_pix()` should be introduced as well.

**Comment by cdeil** [comment]
It turns out this method was used by reproject. I've made it private for now in #1649 . Moving this issue to v0.9, as a reminder to look at whether this extra method is really needed, or if it can be improved.

**Comment by adonath** [comment]
Resolved in #1711.
